experimental/Schemes/CartierDivisor.jl

```diff
@@ -174,34 +174,60 @@ function cartier_divisor(IP::AbsProjectiveScheme, f::Union{MPolyDecRingElem, MPo
   return one(ZZ)*effective_cartier_divisor(IP, f)
 end
 
+### Decompostion of an effective Cartier Divisor into irreducible components
+### (specialized variant of associated_points, using pure codimension 1
+###  and taking multiplicities into account)
+@doc raw"""
+    irreducible_decomposition(C::EffectiveCartierDivisor)
+
+Return a `Vector` of pairs ``(I,k)`` corresponding to the irreducible components of ``C``. More precisely,  each ``I`` is a prime  `IdealSheaf` corresponding to an irreducible component of ``C`` and ``k``is the multiplicity of this component in ``C``.
+"""
+function irreducible_decomposition(C::EffectiveCartierDivisor)
+  cov = trivializing_covering(C)
+  X = scheme(C)
+  OOX = OO(X)
+
+  charts_todo = copy(patches(cov))
+  I = ideal_sheaf(C)
+  associated_primes_temp = Vector{Tuple{IdealSheaf, Int}}()  ## already identified components
+
+  # run through all charts and collect further irreducible components
+  while length(charts_todo) > 0
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                                ## supp(C) might not meet all charts
+    I_temp=I(U)
+
+    for (J,_) in associated_primes_temp
+      !is_one(J(U)) || continue
+      I_temp=saturation(I_temp,J(U))                         ## kick out known components
```

Comment:
It seems that you are not using that any of these ideals is principal, are you? Then I suggest to use `default_covering` instead.

Reply:
Wasn't it that having a principal ideal only pays of in the second argument? Then this seems to almost never be the case here because `J(U)` will usually be constructed by passing from another chart via a saturation process. I'm afraid that the use of the `trivializing_covering` then does not really give any speedup.

Reply:
That was the heart of my question: does trivializing_covering make sense. I agree with your argument for no. So I will change this.

Reply:
I would even take `default_covering` of the scheme. I'm not sure, but `default_covering` of the divisor `C` might also refer to a refinement on which `C` was first defined.

Reply:
done